File: Medium/24. Swap Nodes in Pairs/Solution.java
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head == null) return null;
        ListNode ret = new ListNode(0);
        ret.next = head;
        ListNode first = head;
        ListNode prev = ret;

        while(first != null && first.next != null) {
            ListNode go = first.next.next;
            ListNode second = first.next;

            first.next = go;
            second.next = first;
            prev.next = second;

            prev = first;
            first = go;
        }

        return ret.next;
    }
}

File: Medium/24. Swap Nodes in Pairs/Solution.py
class Solution(object):
    def swapPairs(self, head):
        if head is None:
            return None
        ret = ListNode(0)
        ret.next = head
        first = head
        prev = ret

        while first and first.next:
            go = first.next.next
            second = first.next

            first.next = go
            second.next = first
            prev.next = second

            prev = first
            first = go

        return ret.next